infra/modules/azure_function_app/function_app.tf

```diff
@@ -50,8 +50,16 @@ resource "azurerm_linux_function_app" "this" {
     length(local.function_app.name) > 32 && !(contains(keys(var.app_settings), "AzureFunctionsWebHost__hostid")) ? { AzureFunctionsWebHost__hostid = "production" } : {},
     var.app_settings,
     local.application_insights.enable ? {
+      # SDK AI Sampling (for dependencies)
       # https://docs.microsoft.com/en-us/azure/azure-monitor/app/sampling
-      APPINSIGHTS_SAMPLING_PERCENTAGE = var.application_insights_sampling_percentage
+      APPINSIGHTS_SAMPLING_PERCENTAGE = var.application_insights_sampling_percentage,
+
+      # Runtime AI Sampling (for requests, traces, metrics, etc..)
+      # https://learn.microsoft.com/en-us/azure/azure-functions/configure-monitoring?tabs=v2#overriding-monitoring-configuration-at-runtime
+      AzureFunctionsJobHost__logging__applicationInsights__samplingSettings__maxSamplingPercentage     = var.application_insights_sampling_percentage,
+      AzureFunctionsJobHost__logging__applicationInsights__samplingSettings__initialSamplingPercentage = var.application_insights_sampling_percentage,
+      AzureFunctionsJobHost__logging__logLevel__default                                                = "Information"
+      AzureFunctionsJobHost__logging__logLevel__Function                                               = "Warning"
```

Comment:
this will set at 'Information' (which is somewhat verbose) level even categories like 'Microsoft'. is this expected?

Reply:
"Information" level is the default one, and it's needed for logging requests, metrics and other useful info. See documentation [here](https://learn.microsoft.com/en-us/azure/azure-functions/configure-monitoring?tabs=v2#configure-categories)
That's why we left it as default and we override the Function logLevel.

Reply:
Where in the docs did you find that 'Information' is the default level? I cannot find anything related to default values in the page you've shared. 

What about set "Microsft" to 'Warning' anyway?

Reply:
Where in the docs did you find that 'Information' is the default level? I cannot find anything related to default values in the page you've shared. 

What about set "Microsft" to 'Warning' anyway?

Reply:
> Where in the docs did you find that 'Information' is the default level? I cannot find anything related to default values in the page you've shared.

Actually it's not specified what the default is, but the documentation says that setting the default to `Warning` will not log the requests (Host.Results) neither the metrics (Host.Aggregator).

Anyway I agree that we can set `Microsoft` to `Warning`.

Reply:
@gunzip I'd set `Worker` to `Warning` as well, what do you think?

Reply:
I don't know how this effects traces sent from AI SDK, we should take a look.

Reply:
I thought about this a little bit more. What about stick to defaults for `logLevel`? We may just set sampling percentage with this PR.